packages/angular_devkit/build_angular/src/browser/index.ts

```diff
@@ -231,50 +239,183 @@ export function buildWebpackBrowser(
           1,
         ),
         bufferCount(configs.length),
-        switchMap(buildEvents => {
+        switchMap(async buildEvents => {
+          configs.length = 0;
           const success = buildEvents.every(r => r.success);
-          if (success && options.index) {
+          if (success) {
             let noModuleFiles: EmittedFiles[] | undefined;
             let moduleFiles: EmittedFiles[] | undefined;
             let files: EmittedFiles[] | undefined;
 
-            const [firstBuild, secondBuild] = buildEvents;
-            if (isDifferentialLoadingNeeded) {
-              const scriptsEntryPointName = normalizeExtraEntryPoints(options.scripts || [], 'scripts')
-                .map(x => x.bundleName);
+            const scriptsEntryPointName = normalizeExtraEntryPoints(
+              options.scripts || [],
+              'scripts',
+            ).map(x => x.bundleName);
 
+            const [firstBuild, secondBuild] = buildEvents;
+            if (isDifferentialLoadingNeeded && (fullDifferential || options.watch)) {
               moduleFiles = firstBuild.emittedFiles || [];
-              files = moduleFiles.filter(x => x.extension === '.css' || (x.name && scriptsEntryPointName.includes(x.name)));
+              files = moduleFiles.filter(
+                x => x.extension === '.css' || (x.name && scriptsEntryPointName.includes(x.name)),
+              );
 
               if (buildEvents.length === 2) {
                 noModuleFiles = secondBuild.emittedFiles;
               }
+            } else if (isDifferentialLoadingNeeded && !fullDifferential) {
+              const { emittedFiles = [] } = firstBuild;
+              moduleFiles = [];
+              noModuleFiles = [];
+
+              // Common options for all bundle process actions
+              const actionOptions = {
+                optimize: normalizeOptimization(options.optimization).scripts,
+                sourceMaps:
```

Comment:
Nit: `normalizeSourceMaps(options.sourceMap).scripts`

Reply:
Tries that and it didn’t compile.  Didn’t investigate why though.

Reply:
The problem is that `undefined` is not assignable to 

```js
normalizeSourceMaps(sourceMap: SourceMapUnion)
```

Probably it should be changed to 
```js
export function normalizeSourceMaps(sourceMap: SourceMapUnion = false): SourceMapClass 
```